File: authentication-service/src/main/java/com/microservice/authentication/config/TracerConfiguration.java
package com.microservice.authentication.config;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@ConditionalOnProperty(value = "opentracing.jaeger.enabled", havingValue = "false")
@Configuration
public class TracerConfiguration {

    @Primary
    @Bean
    public io.opentracing.Tracer jaegerTracer() {
        return io.opentracing.noop.NoopTracerFactory.create();
    }
}

File: kotlin-service/src/main/kotlin/com/microservice/kotlin/config/TracerConfiguration.kt
package com.microservice.kotlin.config

import io.opentracing.Tracer
import io.opentracing.noop.NoopTracerFactory
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@ConditionalOnProperty(value = ["opentracing.jaeger.enabled"], havingValue = "false")
@Configuration
class TracerConfiguration {
    @Bean
    fun jaegerTracer(): Tracer? {
        return NoopTracerFactory.create()
    }
}
